httpobs/scanner/analyzer/headers.py

```diff
@@ -137,56 +150,92 @@ def content_security_policy(reqs: dict, expectation='csp-implemented-with-no-uns
+       [source for source in active_csp_sources if 'http:' in source or 'ftp:' in source] and
+       not output['policy']['strictDynamic']):
+        output['result'] = ('csp-implemented-with-insecure-scheme' if output['result'] is None
+                            else output['result'])
+        output['policy']['insecureSchemeActive'] = True
 
     # Don't allow 'unsafe-eval' in script-src or style-src
-    elif script_src.union(style_src).intersection({'\'unsafe-eval\''}):
-        output['result'] = 'csp-implemented-with-unsafe-eval'
+    if script_src.union(style_src).intersection({'\'unsafe-eval\''}):
+        output['result'] = ('csp-implemented-with-unsafe-eval' if output['result'] is None
+                            else output['result'])
+        output['policy']['unsafeEval'] = True
 
     # If the site is https, it shouldn't allow any http: as a source (active content)
-    elif (urlparse(response.url).scheme == 'https' and
-          [source for source in passive_csp_sources if 'http:' in source or 'ftp:' in source]):
-        output['result'] = 'csp-implemented-with-insecure-scheme-in-passive-content-only'
+    if (urlparse(response.url).scheme == 'https' and
+       [source for source in passive_csp_sources if 'http:' in source or 'ftp:' in source]):
+        output['result'] = ('csp-implemented-with-insecure-scheme-in-passive-content-only' if output['result'] is None
+                            else output['result'])
+        output['policy']['insecureSchemePassive'] = True
 
     # Don't allow 'unsafe-inline', data:, or overly broad sources in style-src
-    elif style_src.intersection(DANGEROUSLY_BROAD + UNSAFE_INLINE):
-        output['result'] = 'csp-implemented-with-unsafe-inline-in-style-src-only'
+    if style_src.intersection(DANGEROUSLY_BROAD + UNSAFE_INLINE):
+        output['result'] = ('csp-implemented-with-unsafe-inline-in-style-src-only' if output['result'] is None
+                            else output['result'])
+        output['policy']['unsafeInlineStyle'] = True
 
     # Only if default-src is 'none' and 'none' alone, since additional uris override 'none'
-    elif csp.get('default-src') == {'\'none\''}:
-        output['result'] = 'csp-implemented-with-no-unsafe-default-src-none'
+    if csp.get('default-src') == {'\'none\''}:
+        output['result'] = ('csp-implemented-with-no-unsafe-default-src-none' if output['result'] is None
+                            else output['result'])
+        output['policy']['defaultNone'] = True
     else:
-        output['result'] = 'csp-implemented-with-no-unsafe'
+        output['result'] = ('csp-implemented-with-no-unsafe' if output['result'] is None
+                            else output['result'])
+
+    # Some other checks for the CSP analyzer
+    output['policy']['antiClickjacking'] = False if frame_ancestors.intersection(DANGEROUSLY_BROAD) else True
+    output['policy']['unsafeObjects'] = True if object_src.intersection(DANGEROUSLY_BROAD) else False
```

Comment:
You shouldn't need to use an if/else with booleans, e.g.

```python
output['policy']['antiClickjacking'] = (not frame_ancestors.intersection(DANGEROUSLY_BROAD))
output['policy']['unsafeObjects'] = object_src.intersection(DANGEROUSLY_BROAD)
```

Reply:
`intersection()` produces a set that is the intersection of two sets, it doesn't produce a boolean. Unless I'm misunderstanding you here?

Reply:
Coercing works, I can do that!  :)